### ISSUE TITLE
kwargs passing for connection_type plugins

### ISSUE THREAD

**Issue by nevetS** [ansible/ansible#9788]
##### Issue Type:

“Feature Idea”
##### Ansible Version:

latest
##### Environment:

 “N/A”
##### Summary:

I don't see any documented way to configure connection_type plugins.
##### Steps To Reproduce:

n/a
##### Expected Results:

configurable connection_type plugins
##### Actual Results:

not configurable connection_type plugins

---

I don't see any documented way to configure connection_type plugins.

Most of the existing plugins use an `__init__` call similar to this:

``` python
def __init__(self, runner, host, port, *args, **kwargs):
```

Some of the plugins have custom variables in the ansible.cfg file to define the way they work (`[ssh_connection]->ssh_args` for example).

The ansible.cfg file should be parsed such that any variables listed under `[connectiontype_connection]` are passed as a set of keyword/dict parameters (kwargs) to the plugin.

In other words, if I had this in my ansible.cfg:

```
[my_custom_plugin_connection]
my_custom_executable = /usr/bin/bar
```

I could expect to be able to run

``` python
def __init__(self, runner, host, port, *args, **kwargs):
    if kwargs['my_custom_executable']:
        self.executable = kwargs['my_custom_executable']
    else:
        self.executable = '/usr/bin/foo'
```

I'd be happy to put together a pull request, if this is deemed reasonable.  

If so, a little direction on where the parsing happens, and where the connection_type plugin gets called would be helpful.  (i.e. "have a look at lib/parsers/config-parser.py and lib/runners/connection-plugin-magic.py")

**Comment by cchurch** [comment]
@nevetS I'd proposed https://github.com/ansible/ansible/pull/9601 to allow for easier use of host variables from a connection plugin.  There may be a better way in the works for [v2](https://github.com/ansible/ansible/tree/devel/v2).

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

This has been implemented for some time in 2.0, as well as providing a `set_host_overrides` method to allow connection plugins to change some internal settings based on the destination host (the winrm connection plugin uses this currently).

If you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
